src/Runner.php

```diff
@@ -2,87 +2,94 @@
 
 namespace Pantheon\Terminus;
 
-use Consolidation\AnnotatedCommand\AnnotatedCommandFactory;
 use Consolidation\AnnotatedCommand\CommandFileDiscovery;
+use Consolidation\Log\Logger;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Robo\Robo;
+use Robo\Runner as RoboRunner;
 
 class Runner
 {
     /**
-     * @var Terminus
+     * @var Application
      */
     private $application;
     /**
-     * @var string
+     * @var \Robo\Runner
      */
-    private $commands_directory;
-    /**
-     * @var string
-     */
-    private $top_namespace = 'Pantheon\Terminus\Commands';
+    private $runner;
 
     /**
-     * Runner constructor
+     * Object constructor
      *
-     * @param array $options Options to configure the runner
+     * @param array $options Elements as follow:
+     *        Application application An instance of a Symfony Console Application
+     *        Config      config      A Terminus config instance
      */
-    public function __construct(array $options = [])
+    public function __construct(array $options = ['application' => null, 'config' => null,])
     {
-        $this->commands_directory = __DIR__ . '/Commands';
-        $this->application = new Terminus();
-        $this->configureApplication(new Config($options));
+        $this->application = $options['application'];
```

Comment:
Making good progress here; will have something to push shortly. I am going to check for abstract classes in annotated commands, though, so that we do not have to filter them here. I do think that you should move the TerminusCommand base class to a different namespace anyway, though.